source/core/src/main/com/csse3200/game/components/tasks/ItemProximityTask.java

```diff
@@ -20,7 +21,7 @@ public class ItemProximityTask extends ProximityTask {
     private static final Logger logger = LoggerFactory.getLogger(ItemProximityTask.class);
     private final AbstractItem item;
     private boolean itemPickedUp = false;
-    DialogueBox itemOverlay;
+    private boolean hasApproached;
```

Comment:
This should be inherited as a protected variable from ProximityTask

Reply:
Fixed